pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -467,18 +469,18 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloud.MachineCreateDe
 	awsInstance := &awsInstance{instance: runOut.Instances[0]}
 
 	// Change to our security group
-	_, err = ec2Client.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
+	_, modifyInstanceErr := ec2Client.ModifyInstanceAttribute(&ec2.ModifyInstanceAttributeInput{
 		InstanceId: runOut.Instances[0].InstanceId,
 		Groups:     aws.StringSlice(config.SecurityGroupIDs),
 	})
-	if err != nil {
+	if modifyInstanceErr != nil {
 		_, err := ec2Client.TerminateInstances(&ec2.TerminateInstancesInput{
 			InstanceIds: []*string{runOut.Instances[0].InstanceId},
 		})
 		if err != nil {
-			return nil, awsErrorToTerminalError(err, fmt.Sprintf("failed to attach instance %s to security group id's %v & delete the created instance", aws.StringValue(runOut.Instances[0].InstanceId), config.SecurityGroupIDs))
+			return nil, awsErrorToTerminalError(modifyInstanceErr, fmt.Sprintf("failed to delete instance %s due to %v after attaching to security groups %v", aws.StringValue(runOut.Instances[0].InstanceId), err, config.SecurityGroupIDs))
```

Comment:
@p0lyn0mial It looks like you missed the last four words here

Reply:
the error will be formatted inside the function `awsErrorToTerminalError`. The documentation of that function says that but since you asked we might consider changing the name in the future or change the logic ?

Reply:
ah I see, thanks for the hint. I think the logic is fine. Do you have an idea for a name that makes that behaviour more obvious?

Reply:
the one that says it formats the error, for example `awsErrorToTerminalErrorAndFormat` ?